packages/parse-profile/src/trace/renderEvents.ts

```diff
@@ -46,14 +47,14 @@ function insertRenderEvent(enclosingNode: HierarchyNode<ICpuProfileNode>, event:
   };
 
   // Children who are fully to the left or right of the render event
-  const childrenForOriginal = enclosingNode.children!.filter(child => child.data.max < eventStart ||
+  const childrenForOriginal = getChildren(enclosingNode).filter(child => child.data.max < eventStart ||
```

Comment:
there's an even easier way to do this, taking advantage of the fact that `throw` with a condition has the same effect as using a type guard. Having this precondition at the beginning of your algorithm ensures that your assumptions (i.e., "children are always defined in this particular situation") are not violated, and conveniently leaves you with the types you need.

```ts
const { children } = enclosingNode;

// here children will be of type    undefined | ICpuProfileNode[]

if (!children) throw new Error('Node had undefined children');

// here children will be of type 	ICpuProfileNode[]

// and now the rest of your stuff doesn't require anything special,
// you can use `children` directly with confidence

const childrenForRenderNode = children.filter(child => child.data.min > eventStart && child.data.max < eventEnd);
const leftSplitChild = children.find(n => n.data.min < eventStart && n.data.max > eventStart);
const rightSplitChild = children.find(n => n.data.min < eventEnd && n.data.max > eventEnd);

insertChildInOrder(children, renderNode);
```